### PR TITLE
Rewrite loose equality description

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Motivation

It is so hard to understand (which is understandable, since it's hell complicated). I hope this rewrite makes it clearer.

#### Metadata

- [ ] Adds a new document
- [x] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/equality_comparisons_and_sameness/index.md [hunk 1]
```diff
@@ -61,51 +61,27 @@ Strict equality is almost always the correct comparison operation to use. For al
...
-
-- Loose equality compares two values for equality _after_ converting both values to a common type. After conversions (one or both sides may undergo conversions), the final equality comparison is performed exactly as `===` performs it.
-- Loose equality is _symmetric_: `A == B` always has identical semantics to `B == A` for any values of `A` and `B` (except for the order of applied conversions).
-- `undefined` and `null` are loosely equal; that is, `undefined == null` is true, and `null == undefined` is true
+Loose equality is _symmetric_: `A == B` always has identical semantics to `B == A` for any values of `A` and `B` (except for the order of applied conversions). The behavior for performing loose equality using `==` is as follows:
+
+1. If the operands have the same type, they are compared as follows:
+    - Object: return `true` only if both operands reference the same object.
+    - String: return `true` only if both operands have the same characters in the same order.
+    - Number: return `true` only if both operands have the same value. `+0` and `-0` are treated as the same value. If either operand is `NaN`, return `false`.
```
Comment (teoli2003) [inline files/en-us/web/javascript/equality_comparisons_and_sameness/index.md:69]:
```suggestion
    - Number: return `true` only if both operands have the same value. `+0` and `-0` are treated as the same value. If either operand is `NaN`, return `false`; so `NaN` is never equal to `NaN`.
```

File: files/en-us/web/javascript/reference/operators/equality/index.md [hunk 2]
```diff
@@ -25,41 +25,30 @@ x == y
...
-the [strict equality](/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality) (`===`) operator is
-that the strict equality operator does not attempt type conversion.
-Instead, the strict equality operator always considers operands
-of different types to be different.
+The equality operators (`==` and `!=`) use the [`IsLooselyEqual` Abstract Operation](https://tc39.es/ecma262/#sec-islooselyequal) to compare two operands. This can be roughly summarized as follows:
+
+1. If the operands have the same type, they are compared as follows:
+    - Object: return `true` only if both operands reference the same object.
+    - String: return `true` only if both operands have the same characters in the same order.
+    - Number: return `true` only if both operands have the same value. `+0` and `-0` are treated as the same value. If either operand is `NaN`, return `false`.
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/operators/equality/index.md:33]:
```suggestion
    - Number: return `true` only if both operands have the same value. `+0` and `-0` are treated as the same value. If either operand is `NaN`, return `false`; so, `NaN` is never equal to `NaN`.
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
I think we should explicitly note that `NaN` != `NaN`. But else, looks good!

**Review by teoli2003 — APPROVED** [review]
👍

File: files/en-us/web/javascript/equality_comparisons_and_sameness/index.md [hunk 3]
```diff
@@ -61,51 +61,27 @@ Strict equality is almost always the correct comparison operation to use. For al
...
 
-The behavior for performing loose equality using `==` is as follows:
-
-- Loose equality compares two values for equality _after_ converting both values to a common type. After conversions (one or both sides may undergo conversions), the final equality comparison is performed exactly as `===` performs it.
-- Loose equality is _symmetric_: `A == B` always has identical semantics to `B == A` for any values of `A` and `B` (except for the order of applied conversions).
-- `undefined` and `null` are loosely equal; that is, `undefined == null` is true, and `null == undefined` is true
+Loose equality is _symmetric_: `A == B` always has identical semantics to `B == A` for any values of `A` and `B` (except for the order of applied conversions). The behavior for performing loose equality using `==` is as follows:
+
+1. If the operands have the same type, they are compared as follows:
+    - Object: return `true` only if both operands reference the same object.
```
Comment (hamishwillee) [inline files/en-us/web/javascript/equality_comparisons_and_sameness/index.md:67]: FYI, bullets all indented 1 too many. 3 indents in numbered list, 2 in bullet list.

File: files/en-us/web/javascript/reference/operators/equality/index.md [hunk 4]
```diff
@@ -25,41 +25,30 @@ x == y
...
-    either operand is `NaN`, return `false`.
-  - Boolean: return `true` only if operands are both
-    `true` or both `false`.
-
-The most notable difference between this operator and
-the [strict equality](/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality) (`===`) operator is
-that the strict equality operator does not attempt type conversion.
-Instead, the strict equality operator always considers operands
-of different types to be different.
+The equality operators (`==` and `!=`) use the [`IsLooselyEqual` Abstract Operation](https://tc39.es/ecma262/#sec-islooselyequal) to compare two operands. This can be roughly summarized as follows:
```
Comment (hamishwillee) [inline files/en-us/web/javascript/reference/operators/equality/index.md:28]: "is roughly"

**Comment by hamishwillee** [comment]
FWIW  A couple of comments "after the fact"

- this is so much better. Love it.
- I don't like links to the spec in the text, because these are movable. That's why we keep that stuff in the BCD 
- I would have liked to have seen a link to the terms falsy and truthy here somewhere. Perhaps I missed them.

**Comment by Josh-Cena** [comment]
> I don't like links to the spec in the text, because these are movable

Actually, spec links are stable—even if they change the title, the anchor links will be "redirected". So I won't be particularly worried about that

> I would have liked to have seen a link to the terms falsy and truthy here somewhere

We didn't really mention falsy/truthy here: it would more be a topic in, say, `if...else`. I would probably elaborate on the equality reference page later, and I'll consider that.

File: files/en-us/web/javascript/equality_comparisons_and_sameness/index.md (hunk 3 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/equality_comparisons_and_sameness/index.md:67]: Interesting, I see your point—I'll keep that in mind. I just hit tabs in my editor. Once we start using Prettier it should all be auto-formatted.

Comment (teoli2003) [reply files/en-us/web/javascript/equality_comparisons_and_sameness/index.md:67]: The Daily Markdonwlint PR should detect and fix this.

**Pull request merged**